modules/token_collecting.js

```diff
@@ -256,20 +257,10 @@ class TokenCollecting {
     // the deleted tokens stack is too old - clear it and we'll start again.
     this._deletedTokens = []
     let tokenPool = []
-    if (this._isInChristmasPeriod()) {
-      tokenPool = tokenPool.concat(RoomIdentifiers.christmas)
-    }
-    if (this._isInHalloweenPeriod()) {
-      tokenPool = tokenPool.concat(RoomIdentifiers.halloween)
-    }
-    if (this._isInRickrollPeriod()) {
-      tokenPool = tokenPool.concat(RoomIdentifiers.rickroll)
-    }
-    if (this._isInValentinesPeriod()) {
-      tokenPool = tokenPool.concat(RoomIdentifiers.valentines)
-    }
-    if (this._isInLunarNewYearPeriod()) {
-      tokenPool = tokenPool.concat(RoomIdentifiers.lunarNewYear)
+    for (let period in PeriodicBadges) {
+      if (PeriodBadges[period].isHappening(new Date())) {
```

Comment:
What do you think about moving the `new Date()` call out of the loop?

Reply:
Actually I think we can just reuse `current`

Reply:
Sensible. More sensible than redeclaring it under a different name...